test/built-ins/Object/prototype/toString/symbol-tag-array-builtin.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2019 Alexey Shvayka. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-object.prototype.tostring
+description: >
+  Non-string values of `Symbol.toStringTag` property are ignored.
+info: |
+  Object.prototype.toString ( )
+
+  [...]
+  15. Let tag be ? Get(O, @@toStringTag).
+  16. If Type(tag) is not String, set tag to builtinTag.
+  17. Return the string-concatenation of "[object ", tag, and "]".
+features: [Symbol.toStringTag, Symbol.iterator, iterator-helpers]
+---*/
+
+var toString = Object.prototype.toString;
+
+var arrIter = [][Symbol.iterator]();
+var arrIterProto = Object.getPrototypeOf(arrIter);
+
+assert.sameValue(toString.call(arrIter), '[object Array Iterator]');
+
+Object.defineProperty(arrIterProto, Symbol.toStringTag, {value: null});
+assert.sameValue(toString.call(arrIter), '[object Iterator]');
```

Comment:
this line was corrected from Object to Iterator

Reply:
No, the original test was correct. The line right above this one redefines %IteratorPrototype%'s @@toStringTag to be `null`, so it should be `[object Object]`. Please also double check the other changed tests.

Reply:
It redefines ArrayIteratorPrototype, not IteratorPrototype, i believe.

Reply:
Quite right. Sorry, my reasoning was wrong but the original test is still correct.

`Object.defineProperty(arrIterProto, Symbol.toStringTag, {value: null});` sets the value of @@toStringTag to `null`. Steps 14 and 15 of [Object.prototype.toString](https://tc39.es/ecma262/#sec-object.prototype.tostring) are:

15. Let tag be ? [Get](https://tc39.es/ecma262/#sec-get-o-p)(O, [@@toStringTag](https://tc39.es/ecma262/#sec-well-known-symbols)).
16. If tag [is not a String](https://tc39.es/ecma262/#sec-ecmascript-language-types-string-type), set tag to builtinTag.

So if it finds `null` on `arrIterProto`, finds it unusable, and returns `[object Object]`.

Reply:
Updated - now it verifies its Object when unusable, then deletes it, and then verifies it’s Iterator.